### PR TITLE
[io managers] Add dagster/relation_identifier metadata to IO Manager outputs

### PR DESCRIPTION
**Pull request by benpankow**
## Summary

Automatically populate `dagster/relation_identifier` metadata on assets which are emitted by a DbIOManager.

This should hopefully make it easier to pull data from Dagster into external catalogs.

## Test Plan

Unit tests.

## Changelog [New]

`dagster/relation_identifier` metadata is automatically attached to assets which are stored using an IO manager.

### CODE REVIEW

**Comment by benpankow** [comment]
* **#24196** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/24196?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @benpankow and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: python_modules/dagster/dagster/_core/storage/db_io_manager.py [hunk 1]
```diff
@@ -76,6 +77,18 @@ def delete_table_slice(
     @abstractmethod
     def get_select_statement(table_slice: TableSlice) -> str: ...
 
+    @staticmethod
+    @abstractmethod
```
Comment (sryza) [inline python_modules/dagster/dagster/_core/storage/db_io_manager.py:81]: This shouldn't be abstract, right?

**Review by sryza — APPROVED** [review]
Awesome. One comment worth heeding.

File: python_modules/dagster/dagster/_core/storage/db_io_manager.py (hunk 1 above)
Comment (benpankow) [reply python_modules/dagster/dagster/_core/storage/db_io_manager.py:81]: oh yeah good catch, initially was going to have each connection implement this but realized it was unnecessary

**Pull request merged**